File: template.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main(){
  //box := []int{1, 2, 6, 8}
  data, err := os.Open("./employee1.tsv")
  if err != nil{
    panic(err)
  }
  scan := bufio.NewScanner(data)
  scan.Split(bufio.ScanLines)
  for scan.Scan() {
    word := scan.Text()
    block := strings.Fields(word)
    first_name := block[0]
    last_name := block[1]
    position := block[2]
    letter, ex := os.ReadFile("./letter1.txt")
    if ex != nil {
      panic(ex)
    }

    letter1 := string(letter)
    letter1 = strings.Replace(letter1, "{firstname}", first_name, -1)
    letter1 = strings.Replace(letter1, "{lastname}", last_name, -1)
    letter1 = strings.Replace(letter1, "{position}", position, -1)
    fmt.Println(letter1) 
  }
  data.Close()
}

File: template.py
#!/bin/env python3

# user -> u0_a271
# file -> main.py
# date -> Tue-May-2022  12:31:57
def main():
    count = 0
    with open("./employee.tsv", "r") as theFile:
        next(theFile)
        for lines in theFile:
            linex = lines.split()
            firstname = linex[0]
            lastname = linex[1]
            position = linex[2]
            with open("./letter.txt", "r") as letter:
                output = letter.read()
                output = output.replace("{firstname}", firstname)
                output = output.replace("{lastname}", lastname)
                output = output.replace("{position}", position)
            count += 1
            with open(f"output{count}.txt", "w") as final:
                final.write(output)


if __name__ == "__main__":
    main()
